monkey/monkey_island/cc/ui/src/components/report-components/attack/ReportMatrix.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import Checkbox from '../../ui-components/Checkbox';
+import ReactTable from 'react-table';
+import 'filepond/dist/filepond.min.css';
+import '../../../styles/report/ReportAttackMatrix.scss';
+
+class MatrixComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {techniques: this.props.techniques,
+                  schema: this.props.schema};
+  }
+
+  getColumns() {
+    let columns = [];
+    for(let type_key in this.state.schema.properties){
+      if (! this.state.schema.properties.hasOwnProperty(type_key)){
+        continue;
+      }
+      let tech_type = this.state.schema.properties[type_key];
+      columns.push({
+        Header: () => (<a href={tech_type.link} target="_blank">{tech_type.title}</a>),
+        id: type_key,
+        accessor: x => this.renderTechnique(x[tech_type.title]),
+        style: {'whiteSpace': 'unset'}
+      });
+    }
+    return columns;
+  }
+
+  getTableRows() {
+    let rows = [];
+    for (let tech_id in this.state.techniques) {
+      if (this.state.techniques.hasOwnProperty(tech_id)){
+        let technique_added = false;
+        let technique = this.state.techniques[tech_id];
+        for(let row of rows){
```

Comment:
Try to change to `const` (dict is immutable but the values are not, should be OK)